tfchain_bridge/pkg/stellar/stellar.go

```diff
@@ -306,155 +306,154 @@ func (w *StellarWallet) GetKeypair() *keypair.Full {
 	return w.keypair
 }
 
-// mint handler
-type mint func(map[string]*big.Int, hProtocol.Transaction) error
-
-// // refund handler
-// type refund func(context.Context, string, int64, string) error
-
-// MonitorBridgeAccountAndMint is a blocking function that keeps monitoring
-// the bridge account on the Stellar network for new transactions and calls the
-// mint function when a deposit is made
-func (w *StellarWallet) MonitorBridgeAccountAndMint(ctx context.Context, mintFn mint, stellarCursor string) error {
-	transactionHandler := func(tx hProtocol.Transaction) {
-		if !tx.Successful {
-			return
-		}
-		log.Info().Str("hash", tx.Hash).Msg("Received transaction on bridge stellar account")
-
-		effects, err := w.getTransactionEffects(tx.Hash)
-		if err != nil {
-			log.Error().Str("error while fetching transaction effects:", err.Error())
-			return
-		}
+type MintEventSubscription struct {
+	Events []MintEvent
+	Err    error
+}
 
-		asset := w.GetAssetCodeAndIssuer()
+type MintEvent struct {
+	Senders map[string]*big.Int
+	Tx      hProtocol.Transaction
+	Error   error
+}
 
-		for _, effect := range effects.Embedded.Records {
-			if effect.GetAccount() != w.config.StellarBridgeAccount {
-				continue
-			}
+// getAccountDetails gets account details based an a Stellar address
+func (w *StellarWallet) getAccountDetails(address string) (account hProtocol.Account, err error) {
+	client, err := w.getHorizonClient()
+	if err != nil {
+		return hProtocol.Account{}, err
+	}
+	ar := horizonclient.AccountRequest{AccountID: address}
+	account, err = client.AccountDetail(ar)
+	if err != nil {
+		return hProtocol.Account{}, errors.Wrapf(err, "failed to get account details for account: %s", address)
+	}
+	return account, nil
+}
 
-			if effect.GetType() != "account_credited" {
-				continue
-			}
+func (w *StellarWallet) StreamBridgeStellarTransactions(ctx context.Context, mintChan chan MintEventSubscription, cursor string) error {
+	defer close(mintChan)
```

Comment:
Similar to the substrate wallet, I feel that it should be the channel owner which closes it.